snet_cli/commands.py

```diff
@@ -504,3 +506,212 @@ def transact(self):
             return receipt, events
         else:
             self._error("Cancelled")
+
+
+class ServiceCommand(ClientCommand):
+
+    def _tar_imports(self, paths):
+        max_depth = 0
+        deepest_path = PurePath()
+        for path in paths:
+            path_depth = len(path.parts)
+            if not max_depth or path_depth < max_depth:
+                max_depth = path_depth
+                deepest_path = path
+
+        paths = sorted(paths, key=lambda unsorted_path: (len(unsorted_path.parts), str(unsorted_path.name)))
+        f = tempfile.NamedTemporaryFile()
+        tar = tarfile.open(fileobj=f, mode="w")
+        for path in paths:
+            tar.add(path, path.name) if path == deepest_path else tar.add(path, path.relative_to(deepest_path.parent))
+            self._printout("path: {}".format(str(path)))
+
+        tar.close()
+        return f
+
+    def init(self):
+        accept_all_defaults = self.args.y
+        init_args = {
+            "name": os.path.basename(os.getcwd()),
+            "model": "model/",
```

Comment:
Changed: now we have these options:
- You only provide a directory and entry path will be <directory><service name>.proto
- You provide <path to .proto entry file>#<name of the service>
- You provide a file with no #: only one service must be in the file at compile time.

On a side note, we concurred that "model" must be published with the metadata, otherwise codegen won't know which is the entry file or which service to compile if there are multiple services

Reply:
if you only provide a directory, what happens?

- You scan all files for a service, and if there's more than 1, error?
- You assume all .proto files at the top level of the directory need to be packaged + their transitive dependencies (do you verify there is a service defined in this case?)?

This is hard to understand. IMO,

option 1: model is a path to a proto file. this proto file must contain a single service. we will package the transitive dependencies of this file.
option 2: model is a path to a proto file + #serviceName. we will package the transitive dependencies of this file.

Ideally, we'd enforce always having a #serviceName so that any consumers of this metadata/model (such as the codegen) can reason about the service name without parsing the files again.

other suggestions welcomed but currently the expectations/behavior aren't clear at all.

Reply:
also why didn't this get moved to the metadata as you point out

Reply:
Changed according to the latest discussions: it's always a folder, we don't know if we'll ever need the service name so it is not specified and it does not get published to the blockchain